hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconPipelineReportHandler.java

```diff
@@ -63,13 +64,17 @@ protected void processPipelineReport(PipelineReport report,
     PipelineID pipelineID = PipelineID.getFromProtobuf(report.getPipelineID());
     if (!reconPipelineManager.containsPipeline(pipelineID)) {
       LOG.info("Unknown pipeline {}. Trying to get from SCM.", pipelineID);
-      Pipeline pipelineFromScm = null;
+      Pipeline pipelineFromScm;
 
       try {
         pipelineFromScm =
             scmServiceProvider.getPipeline(report.getPipelineID());
-      } catch (PipelineNotFoundException pnfe) {
-        LOG.error("Could not find pipeline at SCM: {}.", pipelineID, pnfe);
+      } catch (IOException ex) {
+        assert (ex instanceof RemoteException);
+        IOException exception = ((RemoteException) ex)
+                .unwrapRemoteException(PipelineNotFoundException.class);
+        assert (exception instanceof PipelineNotFoundException);
```

Comment:
Not understood why we need this assert here. It can also be any other exception right? (Lets say when SCM's are down)

Reply:
yea, thanks , i made a mistake here, i will change this

Reply:
@bharatviswa504  PTAL, thanks!